src/ipld/util.rs

```diff
@@ -415,7 +420,9 @@ impl<DB: Blockstore + Send + Sync + 'static, T: Borrow<Tipset>, ITER: Iterator<I
                                 if let Some(data) = db.get(&cid)? {
                                     if cid.codec() == fvm_ipld_encoding::DAG_CBOR {
                                         let mut new_values = extract_cids(&data)?;
-                                        cid_vec.append(&mut new_values);
+                                        if !new_values.is_empty() {
+                                            cid_vec.append(&mut new_values);
```

Comment:
Do you think `cid_vec` could grow very large since we are also sending the `state_root` ?

Should we have batch processing here?

Something like this:

```rust=
const MAX_CID_QUEUE_SIZE: usize = 1000;

if cid_vec.len() > MAX_CID_QUEUE_SIZE {
    for cid in cid_vec.drain(MAX_CID_QUEUE_SIZE..) {
        let _ = extract_sender.send(cid);
    }
}
```